carla/recourse_methods/catalog/dice/model.py

```diff
@@ -0,0 +1,83 @@
+import dice_ml
+import pandas as pd
+
+from ...api import Recourse_Method
+
+
+class Dice(Recourse_Method):
+    def __init__(self, mlmodel, data):
+        """
+        Constructor for Dice model
+        Implementation can be seen at https://github.com/interpretml/DiCE
+
+        Restrictions:
+        ------------
+        -   Only the model agnostic approach (backend: sklearn) is used in our implementation.
+        -   ML model needs to have a transformation pipeline for normalization, encoding and feature order.
+            See pipelining at carla/models/catalog/catalog.py for an example ML model class implementation
+
+        Parameters
+        ----------
+        mlmodel : models.api.MLModel
+            ML model to build counterfactuals for.
+        data : data.api.Data
+            Underlying dataset we want to build counterfactuals for.
+        """
+        # Prepare data for dice data structure
+        self._dice_data = dice_ml.Data(
+            dataframe=data.raw,
+            continuous_features=data.continous,
+            outcome_name=data.target,
+        )
+
+        # Build dice model structure
+        # Since our own model class MLModel resembles the sklearn structure it may suffice to use the sklearn
+        # backend implementation made by dice
```

Comment:
Its usually recommended to only have inline comments if an implementation is expectionally counterintuitive for the reader. Or rather comment the "why" in these cases, not the "how". Otherwise we can expect from the reader to read the source code. So in this case here I don't think we neeed these inline comments

Reply:
:+1: